[docs] Add `theme.applyStyles()` documentation with a codemod (#40963)
mui/material-ui: docs/data/material/customization/dark-mode/dark-mode.md

We recommend using this function in favor of the `theme.palette.mode` check, as it's more readable and lets you smoothly adopt new features in the future.

### Usage

With the `**styled**` function:

```jsx
import { styled } from '@mui/material/styles';

const MyComponent = styled('div')(
  ({ theme }) => ({
    color: '#fff',
    backgroundColor: theme.palette.primary.main,
    '&:hover': {
      boxShadow: theme.shadows[3],
      backgroundColor: theme.palette.primary.dark,
    },
  }),
  ({ theme }) =>
    theme.applyStyles('dark', {
      backgroundColor: theme.palette.secondary.main,
    }),
);
```

With the `**sx**` prop:

```jsx
import Button from '@mui/material/Button';

<Button
  sx={[
    (theme) => ({
      color: '#fff',
      backgroundColor: theme.palette.primary.main,
      '&:hover': {
        boxShadow: theme.shadows[3],
        backgroundColor: theme.palette.primary.dark,
      },
    }),
    (theme) =>
      theme.applyStyles('dark', {
        backgroundColor: theme.palette.secondary.main,
      }),
  ]}
>
  Submit
</Button>;
```

### Codemod

> We provide a codemod to migrate your codebase from using `theme.palette.mode` to use `theme.applyStyles()`.

Reviewer: ```suggestion
You can use a codemod to migrate your codebase from using `theme.palette.mode` to use `theme.applyStyles()`.
```
